shell/platform/embedder/tests/embedder_unittests.cc

```diff
@@ -4207,5 +4207,69 @@ TEST_F(EmbedderTest, CompositorRenderTargetsAreInStableOrder) {
   latch.Wait();
 }
 
+TEST_F(EmbedderTest, InvalidAOTDataSourcesMustReturnError) {
+  FlutterEngineAOTDataSource data_in;
+  FlutterEngineAOTData data_out;
+
+  // Invalid FlutterEngineAOTDataSourceType type specified.
+  ASSERT_EQ(FlutterEngineCreateAOTData(&data_in, &data_out), kInvalidArguments);
```

Comment:
Add a test assertion for nullptr inputs.

Reply:
And maybe a check the asserts that the arguments remain unchanged on non-kSuccess returns.